src/transformers/pipelines/text_to_audio.py

```diff
@@ -200,7 +200,8 @@ def _sanitize_parameters(
 
     def postprocess(self, waveform):
         output_dict = {}
-
+        if isinstance(waveform, tuple):
```

Comment:
It would probably make more sense to use `return_dict=True` to make sure we rely on naming rather than indexing? 
This supposed the first is always a waveform

Reply:
not return_dict for semaless_m4v case see https://github.com/huggingface/transformers/blob/main/src/transformers/models/seamless_m4t/modeling_seamless_m4t.py#L3520. also if the class is returned. postprocess in text_to_audio.py is lack of disposal of class input.

Reply:
This is a shortcoming of the modeling code of M4T (and M4Tv2) that doesn't use `return_dict=True`: https://github.com/huggingface/transformers/blob/416711c3ea88109cf25a9c5f85b4aeee2cb831b5/src/transformers/models/seamless_m4t/modeling_seamless_m4t.py#L3520-L3528

Would you add to add `return_dict` to M4T modeling code @sywangyi ? This would look like something like this where relevant: https://github.com/huggingface/transformers/blob/33288ff15011ad4291effa3f1e4912acecc24399/src/transformers/models/roc_bert/modeling_roc_bert.py#L1053-L1063

If not, I can take care of this, but the current PR relies on this fix to have batch_size > 2 working

Reply:
Hi, @ylacombe actually **return_dict_in_generate** should be use. because this is used in generate() func. speech t5 also has such issue. see the return of speech t5. tuple or tensor will be returned. https://github.com/huggingface/transformers/blob/main/src/transformers/models/speecht5/modeling_speecht5.py#L2561-L2593

Reply:
so I think pipeline should support all format output like dict, tuple, tensor, WDYT?